jormungandr-lib/src/interfaces/vote/mod.rs

```diff
@@ -226,14 +338,39 @@ impl From<vote::Payload> for Payload {
             vote::Payload::Public { choice } => Self::Public {
                 choice: choice.as_byte(),
             },
+            vote::Payload::Private {
+                encrypted_vote,
+                proof,
+            } => Self::Private {
+                encrypted_vote: encrypted_vote.iter().flat_map(|ct| ct.to_bytes()).collect(),
+                // TODO: Use iter instead of bytes here
+                proof: ByteBuilder::<u8>::new()
+                    .u64(proof.len() as u64)
```

Comment:
This does not need `u64`, it's `log2(n_proposals)`. In mockchain, the size is furthermore derived from the length of the vote array.

It's better to reuse the `serialize` implementation for `Proof`, which unfortunately we did not define separately.

Reply:
I couldn't find the serialize implementation after some of the changes in chain-libs. Will do better anyway.